baseapp/baseapp.go

```diff
@@ -73,9 +75,10 @@ var _ abci.Application = (*BaseApp)(nil)
 
 // Create and name new BaseApp
 // NOTE: The db is used to store the version number for now.
-func NewBaseApp(name string, cdc *wire.Codec, logger log.Logger, db dbm.DB) *BaseApp {
+func NewBaseApp(name string, cdc *wire.Codec, ctx *sdk.ServerContext, db dbm.DB) *BaseApp {
```

Comment:
@ebuchman and I were discussing the potential of using "functional options" to allow us config flexibility but also allow us to use sane and necessary defaults. Are we're okay with adding these extensible configurations to `sdk.ServerContext`?

Ref: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
e.g. of configurations: a writer for store tracing, pruning configs, txDecoder, etc...

Reply:
This is very interesting. I could see using this, but we would probably just scrap the server context entirely and have BaseApp take these functional parameters, could basically take out everything but name.

Reply:
That is 100% what I was thinking as well.